src/Dialog/Dialog.tsx

```diff
@@ -314,17 +314,18 @@ const Header = styled.div.attrs<SxProp>({as: 'header'})`
   flex-shrink: 0;
 `
 
-const Title = styled.div<SxProp>`
+const Title = styled.h1<SxProp>`
   font-size: ${get('fontSizes.1')};
   font-weight: ${get('fontWeights.bold')};
-
+  margin: 0; /* override default margin */
   ${sx};
 `
 
-const Subtitle = styled.div<SxProp>`
+const Subtitle = styled.h2<SxProp>`
```

Comment:
Wondering if this makes semantic sense. Should it be a `p` instead?

Reply:
Following it from https://github.com/primer/react/issues/1753, should be h2

Reply:
Hmm, I've been thinking about this and it's a good question. I naturally leaned towards `<h2>` but I'm not sure that's actually the right thing. Maybe @bolonio will have opinions here?

Sorry if that turns out to be the wrong tag for the description!

Reply:
First of all, we can agree that Dialog title headings should be `<h1>`. I would go with `<h2>` for the subtitle as well, since modal dialogs are considered areas that are distinct from their parent document, so they have their own heading level structure. I wouldn't say that using a `<p>` is wrong, since not having a `<h2>` heading is possible. I've been reading about it and I couldn't find anything about subtitle or `<h2>` in dialog (of course there are some information about the dialog title  to be `<h1>`). 

Related issues:
- https://github.com/github/accessibility/issues/450#issuecomment-988037997